<a href="https://colab.research.google.com/github/dkumui/IPSD-Assignment/blob/main/MODUL%203%20-%20Visualization/Unguided.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Mengidentifikasi data 'train_features' Guided dan Unguided apakah terdapat perbedaan karakteristik?

In [1]:
import pandas as pd

guided_data = pd.read_csv('train_features (Guided).csv')
unguided_data = pd.read_csv('test_features (Unguided).csv')

guided_info = guided_data.info()
unguided_info = unguided_data.info()

guided_preview = guided_data.head()
unguided_preview = unguided_data.head()

(guided_info, guided_preview, unguided_info, unguided_preview)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3817 entries, 0 to 3816
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   tahun_kelahiran          3817 non-null   int64  
 1   pendidikan               3628 non-null   object 
 2   status_pernikahan        3605 non-null   object 
 3   pendapatan               3627 non-null   float64
 4   jumlah_anak_balita       3627 non-null   float64
 5   jumlah_anak_remaja       3613 non-null   float64
 6   terakhir_belanja         3645 non-null   float64
 7   belanja_buah             3636 non-null   float64
 8   belanja_daging           3639 non-null   float64
 9   belanja_ikan             3624 non-null   float64
 10  belanja_kue              3603 non-null   float64
 11  pembelian_diskon         3639 non-null   float64
 12  pembelian_web            3652 non-null   float64
 13  pembelian_toko           3648 non-null   float64
 14  keluhan                 

(None,
    tahun_kelahiran pendidikan status_pernikahan   pendapatan  \
 0             1979    Sarjana   Rencana Menikah          NaN   
 1             1950    Sarjana   Rencana Menikah   84063000.0   
 2             1966    Sarjana           Menikah  127532564.0   
 3             1961   Magister   Rencana Menikah  165579620.0   
 4             1970    Sarjana   Rencana Menikah  117703159.0   
 
    jumlah_anak_balita  jumlah_anak_remaja  terakhir_belanja  belanja_buah  \
 0                 0.0                 1.0               NaN       50575.0   
 1                 NaN                 NaN              70.0        6069.0   
 2                 0.0                 0.0              45.0      117611.0   
 3                 0.0                 0.0              90.0      206346.0   
 4                 1.0                 1.0              78.0       90563.0   
 
    belanja_daging  belanja_ikan  belanja_kue  pembelian_diskon  pembelian_web  \
 0        260967.0       50575.0      20230.0    

Berdasarkan data di atas, berikut beberapa perbedaan karakteristik antara data train (guided) dan data test (unguided):

1. **Jumlah Kolom:**
   - Data train (guided) memiliki 16 kolom.
   - Data test (unguided) memiliki 17 kolom, dengan tambahan kolom "ID" yang tidak ada di data train.

2. **Jumlah Entri:**
   - Data train memiliki 3,817 entri.
   - Data test memiliki 3,818 entri.

3. **Ketersediaan Data (Missing Values):**
   - Data train memiliki beberapa kolom dengan nilai yang hilang seperti "pendidikan," "status_pernikahan," dan "pendapatan," di mana jumlah entri non-null berbeda dengan jumlah total entri.
   - Data test juga menunjukkan pola missing values yang serupa, tetapi jumlah nilai yang hilang sedikit berbeda untuk kolom-kolom tersebut.

4. **Tanggal Menjadi Anggota:**
   - Kedua dataset memiliki nilai yang sangat sedikit untuk kolom "tanggal_menjadi_anggota" dibandingkan dengan jumlah total entri, menunjukkan banyak nilai yang hilang dalam kolom ini.

Perbedaan ini menunjukkan bahwa meskipun kedua dataset memiliki karakteristik yang mirip, ada perbedaan kecil dalam distribusi nilai yang hilang serta adanya kolom tambahan di data test (unguided).